lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1460,10 +1460,12 @@ namespace Js
     {
         while (JavascriptOperators::GetTypeId(instance) != TypeIds_Null)
         {
-            if (instance->HasProperty(propertyId))
+            PropertyQueryFlags result = instance->HasPropertyQuery(propertyId);
+            if (result != Property_NotFound)
             {
-                return true;
+                return JavascriptConversion::PropertyQueryFlagsToBoolean(result); // return false if instance is typed arrary and HasPropertyQuery() returns Property_Found_Undefined
```

Comment:
Good catch.

---
In reply to: [106293996](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106293996) [](ancestors = 106293996)